tools/evaluate-sroie.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+
+from argparse import ArgumentParser
+from pathlib import Path
+from subprocess import run
+import datasets
+import time
+import tempfile
+from tqdm import tqdm
+
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.metrics import f1_score, precision_score, recall_score
+
+try:
+    import pytesseract
+except ImportError:
+    pytesseract = None
+
+
+def build_ocrs() -> None:
+    run("cargo build --release -p ocrs-cli", shell=True, check=True, text=True)
+
+
+def extract_text(image_path: str) -> str:
+    """Extract text from an image using ocrs."""
+    result = run(
+        # We run the binary directly here rather than use `cargo run` as it
+        # is slightly faster.
+        [f"target/release/ocrs", image_path],
+        check=True,
+        text=True,
+        capture_output=True,
+    )
+    return result.stdout
+
+
+def run_global_retrieval_eval(max_samples: int) -> bool:
+    """
+    Evaluate OCR performance, by computing precision, recall and F1 score
+    for the detected tokens globally on the whole document
```

Comment:
To clarify, is this intended to follow the evaluation protocol from "Scanned Receipt OCR" in the SROIE paper - https://arxiv.org/pdf/2103.10213.pdf? Are there any differences between the tokenization mentioned there and what scikit-learn does as far as you know?

Reply:
So initially I wanted to benchmark datasets, in which case matching exactly the same evaluation procedure is a bit harder. For instance, https://huggingface.co/datasets/naver-clova-ix/cord-v2 should also be easy to add to this script using the same loader. However since SROIE already takes up to 5min (orcs + tesseract) in the end I didn't.

In SROIE they use whitespace tokenization, which is a bit less forgiving than scikit-learn's one. Here is an example,
```py
>>> import re
>>> line = "l'b about 22.33 10/28"
>>> re.findall(r'[^\s]+', line)  # whitespace tokenization
["l'b", 'about', '22.33', '10/28']
>>> re.findall(r'(?u)\b\w\w+\b', line)  # scikit-learn tokenization
['about', '22', '33', '10', '28'] 
```
so you are right's let's revert back to whitespace tokenization as SROIE does, in which case the scores are bit worse,
```
Evaluating on SROIE 2019 dataset...
 - Ocrs: 1.57 s / image, precision 0.88, recall 0.72, F1 0.79
 - Tesseract: 0.86 s / image, precision 0.28, recall 0.26, F1 0.27
```